packages/core/src/sourceTransforms.js

```diff
@@ -12,9 +12,11 @@ function applySourceTransforms(source) {
   ])
 }
 
+// create the RegExp once to avoid creating it over and over again
+// helps work around https://github.com/MetaMask/metamask-extension/issues/21006
+const someDirectEvalPattern = new RegExp('\\beval(\\s*\\()', 'g')
```

Comment:
this looks like it should fail linting because L19 would need to move here as well.

but! why do we need to use a regex literal here?

Reply:
- I don't see why we'd need a literal
- the only concern with a `/g` regex is it maintians an offset marker, so there's one pattern of usage that will not work if the regex is cached. 

Tests might not be complex enough to surface the issue. I'd rather go and refresh my memory of the behavior from documentation and see if the usage pattern is somewhere here below.

If we want to cover for it with blunt force, ` regexp.lastIndex = 0` on line 17 would do it I think.

Reply:
Sorry, I should have wrote "why _can't_ we use a literal"